src/test/java/bio/terra/service/dataset/DatasetBucketDaoTest.java

```diff
@@ -268,34 +273,40 @@ public void BucketMustExistToLink() throws Exception {
     datasetBucketDao.createDatasetBucketLink(datasetId, randomBucketResourceId);
   }
 
-  private UUID createMinimalDataset() throws Exception {
+  private Dataset createMinimalDataset() throws IOException {
     DatasetRequestModel datasetRequest =
         jsonLoader.loadObject("dataset-minimal.json", DatasetRequestModel.class);
     String newName = datasetRequest.getName() + UUID.randomUUID();
     datasetRequest
         .name(newName)
         .defaultProfileId(billingProfile.getId())
         .cloudPlatform(CloudPlatform.GCP);
-    dataset = DatasetUtils.convertRequestWithGeneratedNames(datasetRequest);
-    dataset.projectResourceId(projectId);
+    Dataset dataset = DatasetUtils.convertRequestWithGeneratedNames(datasetRequest);
+    dataset.projectResourceId(projectResource.getId());
     String createFlightId = UUID.randomUUID().toString();
     UUID datasetId = UUID.randomUUID();
+    datasetIds.add(datasetId);
     dataset.id(datasetId);
     datasetDao.createAndLock(dataset, createFlightId);
     datasetDao.unlockExclusive(dataset.getId(), createFlightId);
-    return datasetId;
+
+    return dataset;
   }
 
-  private UUID createBucket() throws InterruptedException {
+  private UUID createBucketDbEntry(GoogleProjectResource projectResource2) {
     String ingestFileFlightId = UUID.randomUUID().toString();
-    GoogleBucketResource bucketForFile =
-        googleBucketService.getOrCreateBucket(
-            gcsConfiguration.getBucket(),
-            projectResource,
-            (GoogleRegion)
-                dataset.getDatasetSummary().getStorageResourceRegion(GoogleCloudResource.BUCKET),
-            ingestFileFlightId,
-            null);
-    return bucketForFile.getResourceId();
+    UUID googleBucketResourceId =
+        resourceDao
+            .createAndLockBucket(
+                String.format("testbucket%s", ingestFileFlightId),
+                projectResource2,
+                (GoogleRegion)
+                    dataset
+                        .getDatasetSummary()
+                        .getStorageResourceRegion(GoogleCloudResource.BUCKET),
+                ingestFileFlightId)
+            .getResourceId();
+    bucketResourceIds.add(googleBucketResourceId);
+    return googleBucketResourceId;
```

Comment:
[Note for code reviewers]: By making this only a dao call, I could switch the whole test file to be unit tests instead of connected tests.